### PR TITLE
Update UI for collections index page; collections listing

### LINKED ISSUE #1509: Collections: Update UI/UX of the listing of collections in the All/My Collections index
### Descriptive summary

Update code that creates the listing of collections in the index view.

NOTE: This issue was extracted from the uber issue for index view changes.  #1417

### Rationale

See model issue for rationale.

### Implementation

**Edit Files:**
* index for All/Your Collections - [app/views/hyrax/my/collections/index.html.erb](https://github.com/samvera/hyrax/blob/master/app/views/hyrax/my/collections/index.html.erb)
* all other files are under
  * [app/views/hyrax/my/collections](https://github.com/samvera/hyrax/blob/master/app/views/hyrax/my/collections) - specific to collections
  * [app/views/hyrax/my](https://github.com/samvera/hyrax/blob/master/app/views/hyrax/my) - shared with works

#### Expected user interactions

Update the UI/UX of the listing of collections in the index views. 

From the All Collections and My Collections pages, a user can...
- [ ] see a list of existing collections and admin sets (see note below on AdminSets)
  - [x] the table for the listing should include columns... 
    - [x] Title - thumbnail and collection title
      - [x] for collections, the title links to `/hyrax/dashboard/collections_controller#show` action
      - [ ] for admin sets, the title links to `/hyrax/admin/admin_set#show` action
    - [x] Type - collection type (e.g. Administrative Set, User Collection, Exhibit, etc.) - This is the title of the collection type. #1570 
    - [x] Visibility - visibility of the collection (e.g. public, private, etc.)
    - [ ] Items - a count of sub-collections and works immediately within the collection (does not cascade to add the count of items in sub-collections)  Can creaet a separate ticket for this if complex, or put in a separate PR once the basic changes are committed.
    - [x] Last modified - TODO: Prefer last mod date, but can start out with `Date added` and enter a ticket to move to Last modified 
    - [x] Actions - actions to take on a collection (i.e., View collection, Edit collection, Delete collection, Add collection to another collection)  NOTE: The processing of actions are being added by other issues.
    
Exact text for collection list and action menu are in the mockup.

#### SPECIAL NOTE on Admin Sets:

For the initial implementation, the Admin Set underlying code will not change.  But to a user, the Admin Set will appear to be a collection.

To make the illusion happen...
* the search builder needs to retrieve both collections and admin sets to be displayed in the list of collections
    
### Mockup

See listing of collections and note 3 showing the Action menu

collection-index-end-user

#### Issues related to UI work for nesting

* #1417 original UI epic for admin index
* #1509 admin index - listing of collections - includes description of action menu items
* #1512 process `Add collection to another collection` action + modal for selecting other collection  (This one has the brain dump list of things needed for nesting)
* #1377 relationship partial for edit form

#### Collections original epic issues (mainly organized by UI)

| Collections feature | Issue |
| ------------------------ | ----- |
| Routes                   | #1380 |
| Controller               | # |
| Model                    | #1383 |
| View - Index             | #1417 |
| Index: Collection List   | (this issue) |
| Index: Filter List      | #1511 |
| Index: New Modal         | #1506 |
| Index: Edit Action       | #1508 |
| Index: Delete Action     | #1507 |
| Index: Add as Subcollection Action | #1512 |
| Index: Bulk Actions      | #1510 |
| View - Show - Admin      | #1427 |
| View - Show - Landing    | #1428 |
| View - New/Edit          | #1374 |
| Partial -  Description   | #1375 |
| Partial -  Branding      | #1376 |
| Partial -  Relationships | #1377 |
| Partial -  Sharing       | #1378 |
| Partial -  Discovery     | #1379 |

### PR DESCRIPTION
**Pull request by adamjarling**
Partially Fixes #1509  ; refs #issuenumber

UI element additions to Collections index page, collections list.

This is a partial fix.  The following still needs to be addressed:

1. The table's 'check all' feature needs to be activated.  I think there is a pattern in the application for how this is implemented.  I tried to copy the pattern method (I think `render_check_all`, and it caused a weird bug that broke the page load and took me a while to track down.
2. The 'Items' column values need to be calculated.
3. Not sure if it's an issue, but there is a lot of repeated code between the following files.  Is this intentional?  Perhaps we could DRY this up and use a template to handle the common parts (if there are indeed differences).  
   - `app/views/hyrax/my/collections/_list_collections.html.erb`
   - `app/views/hyrax/dashboard/collections/_list_collections.html.erb`
   - `app/views/hyrax/collections/_list_collections.html.erb`

@samvera/hyrax-code-reviewers

### CODE REVIEW

File: app/assets/stylesheets/hyrax/_collections.scss [hunk 1]
```diff
@@ -117,3 +117,111 @@ button.branding-banner-remove:hover {
...
+    td {
+      text-align: center;
+      &:nth-child(2) {
+        text-align: left;
+      }
+    }
+  }
+
+  // Datatable overwrites
+  .dataTable {
```
[the remark this answers is not in the document]
Comment (adamjarling) [reply app/assets/stylesheets/hyrax/_collections.scss]: This is over riding a dataTable class.  Not under our control, it's a third party library I believe.

File: app/assets/stylesheets/hyrax/_collections.scss [hunk 2]
```diff
@@ -117,3 +117,101 @@ button.branding-banner-remove:hover {
...
+
+    thead {
+      .sorting::after,
```
[the remark this answers is not in the document]
Comment (adamjarling) [reply app/assets/stylesheets/hyrax/_collections.scss]: Same as above regarding dataTable... these are overriding third party definitions.

**Comment by elrayle** [comment]
I don't see in your code changes where this happens, but the flash message after deleting shows the collection id instead of collection title.

**Comment by elrayle** [comment]
After getting clarification about the mockup, there are a few changed needed for this PR...
- [ ] If bulk `Delete collections` button is clicked and there are no selected collections, show popup message `"You must select one or more collections before you can delete them."`
- [ ] In single collection action menu, change label to `"Add this collection to collection"`
- [ ] In single collection action menu, make `"Add this collection to collection"` the last action in the list
- [ ] In single collection action menu, if `@collection.nestable? false` clicking action `"Add this collection to collection"` should show a popup message `"This type of collection does not support nesting so the collection cannot be added to another collection."

File: app/assets/stylesheets/hyrax/_collections.scss [hunk 3]
```diff
@@ -117,3 +117,87 @@ button.branding-banner-remove:hover {
...
+    .panel-title {
+      font-weight: 300;
+    }
+  }
+}
+
+.collections-wrapper {
+  padding-top: 20px;
+
+  #documents {
```
[the remark this answers is not in the document]
Comment (adamjarling) [reply app/assets/stylesheets/hyrax/_collections.scss]: This is overwriting a pre-defined style in the application (`#documents` selector).

**Comment by ggeisler** [comment]
@elrayle Maybe I'm just getting things mixed up, but in ticket #1509 you seemed to agree with my label suggestion in the mockup of `Add collection to another collection` but in your comment above you are saying it should be changed to `Add this collection to collection`?

If you really prefer the latter, I'd be curious to know your rationale.

**Comment by adamjarling** [comment]
@elrayle All updates are made from the list above.  I fixed a few new failing tests the changes created, however there was one that looked as if it had something to do with a model.  I didn't know where to start, so I 'x'd out of the one spec, and referenced it in the ticket for #1509  in the 'To Do' list.

**Comment by elrayle** [comment]
@ggeisler Huh, when trying to summarize, I read your comment too quickly.  Your comment: "I like the wording in my mockup better than Add this collection to collection"  I missed one little work "than".

It wasn't intentional, but I probably missed it because of my own bias.  The label seems long and open to interpretation.

`Add collection to collection` could be add **this** collection to **another** collection OR add **another** collection to **this** collection.

Would you consider a shorter label...
* `Nest in collection` which gets add **this** collection to **another** collection
* `Add nested sub-collection` which gets add **another** collection to **this** collection

I believe the second label corresponds to the text in the modal `Add selected collection to collection` where the collection you select is added as a sub-collection.

I'm inclined to accept this PR as is and open a new issue for tweaking the labeling.  There is a lot of other good work in this PR that I don't want to hold up.

**Comment by ggeisler** [comment]
@elrayle This appears already merged so I suggest we just deal with it later when all the collections work is done and we can look at everything holistically. I'm fine with whatever you went with for now.

This is just an aside, but I am still confused by your comments, since even in your last comment you seem to be saying that I suggested the label be `Add collection to collection`, which was never something I suggested. My suggestion is `Add collection to another collection`, which doesn't seem ambiguous to me (it is an action on a specific collection for which you've opened the menu). I agree it is a bit long, but it avoids using the terms "nest" and "sub-collection", which make sense to those of us working on the application but I think could be confusing to some of the intended audience.

**Pull request merged**